src/main/scala/Column.scala

```diff
@@ -452,6 +463,57 @@ case class Column private (
+    "http://www.w3.org/2001/XMLSchema#boolean" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#date" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#dateTime" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#dateTimeStamp" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#decimal" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#integer" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#long" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#int" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#short" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#byte" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#nonNegativeInteger" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#positiveInteger" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#unsignedLong" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#unsignedInt" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#unsignedShort" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#unsignedByte" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#nonPositiveInteger" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#negativeInteger" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#double" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#duration" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#dayTimeDuration" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#yearMonthDuration" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#float" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#gDay" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#gMonth" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#gMonthDay" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#gYear" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#gYearMonth" -> noAdditionalValidation _,
+    "http://www.w3.org/2001/XMLSchema#hexBinary" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#QName" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#string" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#normalizedString" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#token" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#language" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#Name" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#NMTOKEN" -> regexpValidation _,
+    "http://www.w3.org/2001/XMLSchema#time" -> noAdditionalValidation _
+  )
+
+  def regexpValidation(value: String): Boolean = {
+    format.get.pattern.get.r.pattern.matcher(value).matches()
+  }
```

Comment:
This looks a bit messy and hard to read. Perhaps 
```
val regExPattern = format.flatMap(f => f.pattern)
                                .get
val regEx = regExPattern.r
return r.pattern.matcher(value).matches()
```

It feels like the above variable names help guide me through the process a bit better.

Reply:
👍